### PR TITLE
Common: fix warnings in directory.cpp

### PR DESCRIPTION
**Pull request by ericoporto**
minor warnings and clang-tidy notices

### CODE REVIEW

File: Common/util/directory.cpp [hunk 1]
```diff
@@ -106,15 +106,15 @@ bool GetFilesImpl(const String &dir_path, std::vector<String> &files,
...
+    struct stat f_stat{};
     while ((ent = readdir(dir)) != nullptr)
     {
         if (strcmp(ent->d_name, ".") == 0 ||
             strcmp(ent->d_name, "..") == 0) continue;
         if (fstatat(dir_fd, ent->d_name, &f_stat, 0) != 0) continue;
         if (S_ISREG(f_stat.st_mode) == is_reg &&
             S_ISDIR(f_stat.st_mode) == is_dir)
-            files.push_back(ent->d_name);
+            files.emplace_back(ent->d_name);
```
Comment (ivan-mogilko) [inline Common/util/directory.cpp:117]: There's another instance of `files.push_back` in this code (slightly below), does it also need to be replaced?

Comment (ericoporto) [reply Common/util/directory.cpp:117]: thanks! I was using Linux, now tried on Windows and saw the different code paths. I noticed SetCurrentDirectory, GetDirs and GetFiles are apparently not used.

Comment (ivan-mogilko) [reply Common/util/directory.cpp:117]: > I noticed SetCurrentDirectory, GetDirs and GetFiles are apparently not used.

GetDirs and GetFiles are likely used by tools.

Comment (ericoporto) [reply Common/util/directory.cpp:117]: Ahh, that's right! I forgot those! OK! :)

**Pull request merged**